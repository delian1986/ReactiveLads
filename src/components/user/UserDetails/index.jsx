import { Component } from "react";
import { Redirect } from "react-router-dom";
import PropTypes from "prop-types";
export class UserDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      goBack: false,
      id: props.id,
      email: props.email,
      firstName: props.firstName,
      lastName: props.lastName,
      photoUrl: props.photoUrl,
      password: props.password
    };
  }

  applyChanges = (e) => {
    e.preventDefault();
    this.props.applyChangesThunk(this.state);
    this.navigateBack();
  };

  cancelChanges = () => {
    this.navigateBack();
  };

  navigateBack = () => {
    this.setState({ goBack: true });
  };

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    if (this.state.goBack) {
      return <Redirect />;
    }
    return (
      <div className="container h-100">
        <div className="row h-100 justify-content-center align-items-center">
          <div className="col-6 mx-auto">
            <div className="card card-body bg-light">
              <form onSubmit={this.applyChanges}>
                <h5>User details</h5>
                <input
                  className="form-control"
                  id="1"
                  name="firstName"
                  onChange={this.onChange}
                  type="text"
                  placeholder="First name"
                  defaultValue={this.props.firstName}
                />
                <br />
                <input
                  className="form-control"
                  id="2"
                  name="lastName"
                  onChange={this.onChange}
                  type="text"
                  placeholder="Last name"
                  defaultValue={this.props.lastName}
                />
                <br />
                <input
                  className="form-control"
                  id="3"
                  name="photoUrl"
                  onChange={this.onChange}
                  type="text"
                  placeholder="Photo"
                  defaultValue={this.props.photoUrl}
                />
                <br />
                <input
                  className="form-control"
                  id="4"
                  name="email"
                  onChange={this.onChange}
                  type="text"
                  placeholder="Email"
                  defaultValue={this.props.email}
                />
                <br />
                <input
                  className="form-control"
                  id="5"
                  name="password"
                  onChange={this.onChange}
                  required
                  minLength="4"
                  type="password"
                  placeholder="Password"
                />
                <br />
                <button className="btn btn-block" type="submit" name="Apply">
                  Apply
                </button>
                <button
                  className="btn btn-block"
                  to="/"
                  onClick={this.cancelChanges}
                >
                  Cancel
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

UserDetails.propTypes = {
  applyChangesThunk: PropTypes.func,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  photoUrl: PropTypes.string,
  email: PropTypes.string,
  password: PropTypes.string
};
