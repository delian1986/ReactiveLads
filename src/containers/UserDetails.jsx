import { connect } from "react-redux";
import { UserDetails } from "../components/user/UserDetails/index";
import { bindActionCreators } from "redux";
import { getUserDetailsAsync } from "../store/actions/auth";

const mapStateToProps = (state) => {
  return {
    id: state.auth.id,
    firstName: state.auth.firstName || "",
    lastName: state.auth.lastName || "",
    photoUrl: state.auth.photoUrl || "",
    email: state.auth.email || "",
    password: state.auth.password || ""
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(
    {
      applyChangesAsync: ({ id, firstName, lastName, photoUrl, email, password }) =>
        getUserDetailsAsync({ id, firstName, lastName, photoUrl, email, password })
    },
    dispatch
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(UserDetails);
