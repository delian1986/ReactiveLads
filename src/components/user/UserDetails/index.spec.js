import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import UserDetails from "../../../containers/UserDetails";

const mockStore = configureStore([]);
describe("tests <UserDetails/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      auth: {
        isPending: false
      },
      message: {}
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <UserDetails />
      </Provider>
    );
  });

  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
