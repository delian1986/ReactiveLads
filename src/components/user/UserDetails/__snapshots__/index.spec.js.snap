// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <UserDetails/> component Test rendering component should render with given state from Redux store 1`] = `
<div
  className="container h-100"
>
  <div
    className="row h-100 justify-content-center align-items-center"
  >
    <div
      className="col-6 mx-auto"
    >
      <div
        className="card card-body bg-light"
      >
        <form
          onSubmit={[Function]}
        >
          <h5>
            User details
          </h5>
          <input
            className="form-control"
            defaultValue=""
            id="1"
            name="firstName"
            onChange={[Function]}
            placeholder="First name"
            type="text"
          />
          <br />
          <input
            className="form-control"
            defaultValue=""
            id="2"
            name="lastName"
            onChange={[Function]}
            placeholder="Last name"
            type="text"
          />
          <br />
          <input
            className="form-control"
            defaultValue=""
            id="3"
            name="photoUrl"
            onChange={[Function]}
            placeholder="Photo"
            type="text"
          />
          <br />
          <input
            className="form-control"
            defaultValue=""
            id="4"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <br />
          <input
            className="form-control"
            id="5"
            minLength="4"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
          <br />
          <button
            className="btn btn-block"
            name="Apply"
            type="submit"
          >
            Apply
          </button>
          <button
            className="btn btn-block"
            onClick={[Function]}
            to="/"
          >
            Cancel
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
